icaruscode/PMT/OpReco/ICARUSFlashAssAna_module.cc

```diff
@@ -350,36 +537,97 @@ void opana::ICARUSFlashAssAna::analyze(art::Event const& e) {
   m_timestamp = e.time().timeHigh(); // precision to the second 
 
 
+  /*
+  This part is for the trigger information
+  */
+
+
   // We work out the trigger information here 
   if( !fTriggerLabel.empty() ) { 
 
+      // Beam information
       art::Handle<std::vector<sim::BeamGateInfo>> beamgate_handle;
       e.getByLabel( fTriggerLabel, beamgate_handle );
-
+      
       if( beamgate_handle.isValid() ) {
 
         for( auto const & beamgate : *beamgate_handle ) {
 
           m_beam_gate_start = beamgate.Start(); 
           m_beam_gate_width = beamgate.Width(); 
-          m_beam_type = beamgate.BeamType() ;
+          m_beam_type = beamgate.BeamType() ; 
 
         }
 
       }
 
       else {
-        std::cout << "Invalid Trigger Data product " << fTriggerLabel.label() << "\n" ;
+        if( fDebug) { std::cout << "No sim::BeamGateInfo associated to label: " << fTriggerLabel.label() << "\n" ; }
+      }
+
+      // Now trigger information
+      art::Handle<sbn::ExtraTriggerInfo> trigger_handle;
+      e.getByLabel( fTriggerLabel, trigger_handle );
+
+      if( trigger_handle.isValid() ) {
+
+        sbn::triggerSource bit = trigger_handle->sourceType;
+
+        m_gate_type = (unsigned int)bit; 
+        m_gate_name = bitName(bit);
+        m_trigger_timestamp = trigger_handle->triggerTimestamp;
+        m_gate_start_timestamp =  trigger_handle->beamGateTimestamp;
+        m_trigger_gate_diff = m_trigger_timestamp-m_gate_start_timestamp;
```

Comment:
To make sure that you are not affected by any mistake in the conversion, I suggest you use the original timestamps in this formula:
```suggestion
        m_trigger_gate_diff = trigger_handle->triggerTimestamp - trigger_handle->beamGateTimestamp;
```